homeassistant/components/dsmr/sensor.py

```diff
@@ -381,21 +381,12 @@ def should_poll(self):
         return False
 
     @staticmethod
-    def translate_tariff(value, dsmr_version):
-        """Convert 2/1 to normal/low depending on DSMR version."""
-        # DSMR V5B: Note: In Belgium values are swapped:
-        # Rate code 2 is used for low rate and rate code 1 is used for normal rate.
-        if dsmr_version in ("5B",):
-            if value == "0001":
-                value = "0002"
-            elif value == "0002":
-                value = "0001"
-        # DSMR V2.2: Note: Rate code 1 is used for low rate and rate code 2 is
-        # used for normal rate.
-        if value == "0002":
-            return "normal"
+    def translate_tariff(value):
```

Comment:
This function has no use anymore, converting it to float in the state function should remove the leading zeros.

Reply:
Wanted to handle it as a string and avoid the decimal. But indeed better to process it like the other values.

Reply:
Hmm, this doesn't look good either, a value "1.0" is not really what we want I think, the decimal point is useless.

Perhaps just convert to int for the tariff entity?

It's a number so having it not as string makes sense, then users can treat it as number.

Reply:
The useless .0 is consistent with the 'power failure count' and 'voltage sags' sensors.

Reply:
Yeah, you're right, we should treat them all as int, but that's outside the scope of this PR.

Reply:
Agree, int now.